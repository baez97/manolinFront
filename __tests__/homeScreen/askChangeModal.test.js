import React from 'react';
import renderer from 'react-test-renderer';
import AskChangeModal from '../../components/homeComponents/askChangeModal';

jest.unmock("react-native")
jest.unmock('react')
jest.unmock("react-test-renderer")
jest.unmock('expo');

describe("AskChangeModal", () => {
    const mockedProps = {
        addChange   : jest.fn(),
        addFree     : jest.fn(),
        toggleModal : jest.fn(),
        selectedTurn   : {},
        isModalVisible : true,
    }
    const askModal = new AskChangeModal(mockedProps);

    describe("Methods", () => {
        it("AskChangePressed", () => {
            askModal.askChangePressed();

            expect( mockedProps.addChange ).toBeCalled();
        });

        it("AskChangePressed", () => {
            askModal.askFreePressed();

            expect( mockedProps.addFree ).toBeCalled();
        });
    });

    describe("View", () => {
        const addChange   = jest.fn();
        const addFree     = jest.fn();
        const toggleModal = jest.fn();
        const selectedTurn   = {};
        const isModalVisible = false;

        it("Matches the Snapshot when it is not visible", () =>{
            const view =
                <AskChangeModal
                    addChange      = { addChange }
                    addFree        = { addFree }
                    toggleModal    = { toggleModal }
                    selectedTurn   = { selectedTurn }
                    isModalVisible = { isModalVisible }/>

            const rendered = renderer.create(view);
            const renderedJson = rendered.toJSON();

            expect( renderedJson ).toMatchSnapshot();
        });

        it("Matches the Snapshot when it is visible", () =>{
            const isModalVisible = true;

            const view =
                <AskChangeModal
                    addChange      = { addChange }
                    addFree        = { addFree }
                    toggleModal    = { toggleModal }
                    selectedTurn   = { selectedTurn }
                    isModalVisible = { isModalVisible }/>

            const rendered = renderer.create(view);
            const renderedJson = rendered.toJSON();

            expect( renderedJson ).toMatchSnapshot();
        });
    });
});