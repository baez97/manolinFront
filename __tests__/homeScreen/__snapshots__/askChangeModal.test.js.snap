// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AskChangeModal View Matches the Snapshot when it is not visible 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  hideModalContentWhileAnimating={false}
  onModalHide={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  onRequestClose={[MockFunction]}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  swipeThreshold={100}
  transparent={true}
  visible={false}
>
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "black",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
  />
  <View
    hideModalContentWhileAnimating={false}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    pointerEvents="box-none"
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "margin": 37.5,
        "transform": Array [
          Object {
            "translateY": 0,
          },
        ],
      }
    }
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderRadius": 35.5,
            "flexDirection": "column",
            "justifyContent": "flex-start",
            "paddingBottom": 43.75,
            "paddingLeft": 26.25,
            "paddingRight": 26.25,
            "paddingTop": 26.25,
            "width": 602,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#0470dc",
              "fontFamily": "montserrat-extra-bold",
              "fontSize": 40.7291678090925,
              "marginBottom": 17.5,
              "textAlign": "center",
            }
          }
        >
          undefined undefined de undefined
        </Text>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <ExponentLinearGradient
            colors={
              Array [
                4287123141,
                4281607599,
              ]
            }
            endPoint={
              Array [
                0.8,
                1,
              ]
            }
            startPoint={
              Array [
                0.7,
                0,
              ]
            }
            style={
              Object {
                "alignItems": "center",
                "borderRadius": 35.5,
                "elevation": 15,
                "height": 141.5,
                "justifyContent": "center",
                "marginTop": 26.25,
                "width": 442.5,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontFamily": "montserrat-extra-bold",
                  "fontSize": 38.95833442608516,
                  "textAlign": "center",
                }
              }
            >
              Pedir cambio
            </Text>
          </ExponentLinearGradient>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <ExponentLinearGradient
            colors={
              Array [
                4287123141,
                4281607599,
              ]
            }
            endPoint={
              Array [
                0.8,
                1,
              ]
            }
            startPoint={
              Array [
                0.7,
                0,
              ]
            }
            style={
              Object {
                "alignItems": "center",
                "borderRadius": 35.5,
                "elevation": 15,
                "height": 141.5,
                "justifyContent": "center",
                "marginTop": 26.25,
                "width": 442.5,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontFamily": "montserrat-extra-bold",
                  "fontSize": 38.95833442608516,
                  "textAlign": "center",
                }
              }
            >
              Pedir libre
            </Text>
          </ExponentLinearGradient>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <ExponentLinearGradient
            colors={
              Array [
                4279286482,
                4278481116,
              ]
            }
            endPoint={
              Array [
                0.65,
                1,
              ]
            }
            startPoint={
              Array [
                0.55,
                0,
              ]
            }
            style={
              Object {
                "alignItems": "center",
                "borderRadius": 35.5,
                "elevation": 15,
                "height": 141.5,
                "justifyContent": "center",
                "marginTop": 61.25,
                "width": 442.5,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontFamily": "montserrat-extra-bold",
                  "fontSize": 38.95833442608516,
                  "textAlign": "center",
                }
              }
            >
              Cancelar
            </Text>
          </ExponentLinearGradient>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;

exports[`AskChangeModal View Matches the Snapshot when it is visible 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  hideModalContentWhileAnimating={false}
  onModalHide={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  onRequestClose={[MockFunction]}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  swipeThreshold={100}
  transparent={true}
  visible={true}
>
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "black",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
  />
  <View
    hideModalContentWhileAnimating={false}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    pointerEvents="box-none"
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "margin": 37.5,
        "transform": Array [
          Object {
            "translateY": 1334,
          },
        ],
      }
    }
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderRadius": 35.5,
            "flexDirection": "column",
            "justifyContent": "flex-start",
            "paddingBottom": 43.75,
            "paddingLeft": 26.25,
            "paddingRight": 26.25,
            "paddingTop": 26.25,
            "width": 602,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#0470dc",
              "fontFamily": "montserrat-extra-bold",
              "fontSize": 40.7291678090925,
              "marginBottom": 17.5,
              "textAlign": "center",
            }
          }
        >
          undefined undefined de undefined
        </Text>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <ExponentLinearGradient
            colors={
              Array [
                4287123141,
                4281607599,
              ]
            }
            endPoint={
              Array [
                0.8,
                1,
              ]
            }
            startPoint={
              Array [
                0.7,
                0,
              ]
            }
            style={
              Object {
                "alignItems": "center",
                "borderRadius": 35.5,
                "elevation": 15,
                "height": 141.5,
                "justifyContent": "center",
                "marginTop": 26.25,
                "width": 442.5,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontFamily": "montserrat-extra-bold",
                  "fontSize": 38.95833442608516,
                  "textAlign": "center",
                }
              }
            >
              Pedir cambio
            </Text>
          </ExponentLinearGradient>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <ExponentLinearGradient
            colors={
              Array [
                4287123141,
                4281607599,
              ]
            }
            endPoint={
              Array [
                0.8,
                1,
              ]
            }
            startPoint={
              Array [
                0.7,
                0,
              ]
            }
            style={
              Object {
                "alignItems": "center",
                "borderRadius": 35.5,
                "elevation": 15,
                "height": 141.5,
                "justifyContent": "center",
                "marginTop": 26.25,
                "width": 442.5,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontFamily": "montserrat-extra-bold",
                  "fontSize": 38.95833442608516,
                  "textAlign": "center",
                }
              }
            >
              Pedir libre
            </Text>
          </ExponentLinearGradient>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <ExponentLinearGradient
            colors={
              Array [
                4279286482,
                4278481116,
              ]
            }
            endPoint={
              Array [
                0.65,
                1,
              ]
            }
            startPoint={
              Array [
                0.55,
                0,
              ]
            }
            style={
              Object {
                "alignItems": "center",
                "borderRadius": 35.5,
                "elevation": 15,
                "height": 141.5,
                "justifyContent": "center",
                "marginTop": 61.25,
                "width": 442.5,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontFamily": "montserrat-extra-bold",
                  "fontSize": 38.95833442608516,
                  "textAlign": "center",
                }
              }
            >
              Cancelar
            </Text>
          </ExponentLinearGradient>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
